pkg/gardenlet/controller/managedseed/managedseed_reconciler.go

```diff
@@ -0,0 +1,230 @@
+	conditionShootReconciled := gardencorev1beta1helper.GetOrInitCondition(managedSeed.Status.Conditions, seedmanagementv1alpha1.ManagedSeedShootReconciled)
+	conditionSeedRegistered := gardencorev1beta1helper.GetOrInitCondition(managedSeed.Status.Conditions, seedmanagementv1alpha1.ManagedSeedSeedRegistered)
+	conditionSeedBootstrapped := gardencorev1beta1helper.GetOrInitCondition(managedSeed.Status.Conditions, seedmanagementv1alpha1.ManagedSeedSeedBootstrapped)
+
+	defer func() {
+		if err := updateStatus(ctx, r.gardenClient.Client(), managedSeed, conditionShootExists, conditionShootReconciled, conditionSeedRegistered, conditionSeedBootstrapped); err != nil {
+			logger.Errorf("Could not update status: %+v", err)
+		}
+	}()
+
+	// Ensure the shoot exists and update the ShootExists condition
+	shoot, err := r.ensureShootExists(ctx, managedSeed, &conditionShootExists, logger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if the shoot is reconciled and update the ShootReconciled condition
+	// If the shoot is not reconciled yet, requeue for another check in 10s
+	if shoot.Generation != shoot.Status.ObservedGeneration || shoot.Status.LastOperation == nil || shoot.Status.LastOperation.State != gardencorev1beta1.LastOperationStateSucceeded {
+		r.recordAndLog(managedSeed, fmt.Sprintf("Shoot %s is still reconciling", kutil.ObjectName(shoot)), &conditionShootReconciled,
+			gardencorev1beta1.ConditionFalse, "ShootStillReconciling", corev1.EventTypeNormal, gardencorev1beta1.EventReconciling, logger, logrus.InfoLevel)
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+	r.recordAndLog(managedSeed, fmt.Sprintf("Shoot %s is reconciled", kutil.ObjectName(shoot)), &conditionShootReconciled,
+		gardencorev1beta1.ConditionTrue, "ShootReconciled", corev1.EventTypeNormal, gardencorev1beta1.EventReconciling, logger, logrus.InfoLevel)
+
+	// Reconcile creation or update and update the SeedRegistered condition
+	if err := r.actuator.Reconcile(ctx, managedSeed, shoot); err != nil {
+		r.recordAndLog(managedSeed, fmt.Sprintf("Could not register seed: %+v", err), &conditionSeedRegistered,
+			gardencorev1beta1.ConditionFalse, "SeedRegistrationFailed", corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, logger, logrus.ErrorLevel)
+		return reconcile.Result{}, fmt.Errorf("could not register seed: %w", err)
+	}
+	r.recordAndLog(managedSeed, fmt.Sprintf("Shoot %s registered as seed", kutil.ObjectName(shoot)), &conditionSeedRegistered,
+		gardencorev1beta1.ConditionTrue, "SeedRegistered", corev1.EventTypeNormal, gardencorev1beta1.EventReconciled, logger, logrus.InfoLevel)
+
+	// Check if the seed has been successfully bootstrapped and update the SeedBootstrapped condition
+	// If the seed is not bootstrapped yet, requeue for another check in 10s
+	bootstrapped, err := r.checkSeedBootstrapped(ctx, managedSeed, &conditionSeedBootstrapped, logger)
```

Comment:
Yes, you are right. There are also other arguments in favor of a secondary controller - for example, if the seed has been bootstrapped once, but later loses its "bootstrapped" status (actually observed this a few times during my tests), the `SeedBootstraped` condition in the ManagedSeed won't be updated, since it's not reconciled anymore. Similarly, there is a (usually very short, unless there are errors) time window during deletion during which the seed is no longer bootstrapped but the ManagedSeed is still `Ready`.

On the other hand, I am wary of potential "complexity explosion" if we introduce a secondary controller. I think the impact of the above is very low and we could perhaps simply live with it: 
* To your comment, the registration will be retried, but if there are no changes to the `ManagedSeed` it's essentially a no-op, since exactly the same `Seed` will be created / gardenlet will be deployed. The impact is nil from my perspective.
* The "bootstrapped" status of the Seed and the "ready" status of the ManagedSeed might become out of sync, usually for short periods of time, but the impact is also low since right now the "ready" status of the ManagedSeed is simply a UX feature, it's not used for anything else.

Reply:
We discussed this at length with @timebertt and agreed that the the `SeedBootstrapped` cannot be properly maintained just by the managed seed controller, for the reasons mentioned above. If we do want to have this condition, we should introduce a new controller like suggested above. However, we are both not convinced that this condition is that useful as to warrant the increased complexity from having to implement a new controller.

We should either:
* Completely remove the `SeedBootstrapped` condition, since right now it's just an UX feature. To avoid confusion, we should rename the `Ready` column to something like `Registered`.
* Keep the `SeedBootstrapped` condition and introduce a new "secondary" controller to properly maintain it.

In both cases, the managed seed controller should consider its job done once it has reconciled the managed seed and not wait until the seed is actually "bootstrapped".

I personally vote strongly for the first option right now, we could always introduce the new condition and controller if it becomes really important later.

@danielfoehrKn @rfranzke What do you think? I am sorry for the back and forth, I agreed with introducing the `SeedBootstrapped` condition because I thought it wouldn't be a big deal, however this turned out not to be the case. So unless you have strong objections, I will remove it again.

Reply:
Thanks for your discussion and the summary - very helpful. I agree and also vote for the first option for the arguments you mentioned. I like renaming `Ready` to `Registered` - it makes perfect sense. The `ManagedSeed` will show whether it did it's job which does not necessarily mean that everything is ready for scheduling shoots - this will only be visible in the `Seed` itself. I think that's fair for now given that it's only a UX feature.

Reply:
Removed the `SeedBootstrapped` condition